tests/gtest/avifgainmaptest.cc

```diff
@@ -1217,139 +1232,194 @@ INSTANTIATE_TEST_SUITE_P(
-  const std::string hdr_image_name = "seine_hdr_gainmap_srgb.avif";
-  ImagePtr sdr_image =
-      testutil::DecodeFile(std::string(data_path) + sdr_image_name);
-  ASSERT_NE(sdr_image, nullptr);
-  ImagePtr hdr_image =
-      testutil::DecodeFile(std::string(data_path) + hdr_image_name);
-  ASSERT_NE(hdr_image, nullptr);
+  const std::string image1_name =
+      std::get<0>(GetParam());  //"seine_sdr_gainmap_srgb.avif";
+  const std::string image2_name =
+      std::get<1>(GetParam());  //"seine_hdr_gainmap_srgb.avif";
+  const int downscaling = std::get<2>(GetParam());
+  const int gain_map_depth = std::get<3>(GetParam());
+  const avifPixelFormat gain_map_format = std::get<4>(GetParam());
+  const float min_psnr = std::get<5>(GetParam());
+  const float max_psnr = std::get<6>(GetParam());
+
+  ImagePtr image1 = testutil::DecodeFile(std::string(data_path) + image1_name);
+  ASSERT_NE(image1, nullptr);
+  ImagePtr image2 = testutil::DecodeFile(std::string(data_path) + image2_name);
+  ASSERT_NE(image2, nullptr);
 
   const uint32_t gain_map_width = std::max<uint32_t>(
-      (uint32_t)std::round((float)sdr_image->width / downscaling), 1u);
+      (uint32_t)std::round((float)image1->width / downscaling), 1u);
   const uint32_t gain_map_height = std::max<uint32_t>(
-      (uint32_t)std::round((float)sdr_image->height / downscaling), 1u);
+      (uint32_t)std::round((float)image1->height / downscaling), 1u);
   std::unique_ptr<avifGainMap, decltype(&avifGainMapDestroy)> gain_map(
       avifGainMapCreate(), avifGainMapDestroy);
-  ImagePtr gain_map_image(avifImageCreate(gain_map_width, gain_map_height,
-                                          gain_map_depth, gain_map_format));
-  gain_map->image =
-      gain_map_image.release();  // 'gain_map' now owns gain_map_image;
+  gain_map->image = avifImageCreate(gain_map_width, gain_map_height,
+                                    gain_map_depth, gain_map_format);
 
   avifDiagnostics diag;
-  avifResult result = avifImageComputeGainMap(sdr_image.get(), hdr_image.get(),
+  gain_map->metadata.useBaseColorSpace = true;
+  avifResult result = avifImageComputeGainMap(image1.get(), image2.get(),
                                               gain_map.get(), &diag);
   ASSERT_EQ(result, AVIF_RESULT_OK)
       << avifResultToString(result) << " " << diag.error;
 
   EXPECT_EQ(gain_map->image->width, gain_map_width);
   EXPECT_EQ(gain_map->image->height, gain_map_height);
 
-  const float hdr_headroom = (float)gain_map->metadata.alternateHdrHeadroomN /
-                             gain_map->metadata.alternateHdrHeadroomD;
+  const float image1_headroom = (float)gain_map->metadata.baseHdrHeadroomN /
+                                gain_map->metadata.baseHdrHeadroomD;
+  float image2_headroom = (float)gain_map->metadata.alternateHdrHeadroomN /
+                          gain_map->metadata.alternateHdrHeadroomD;
+
+  if (image1_headroom == image2_headroom) {
+    gain_map->metadata.alternateHdrHeadroomN += 1;
```

Comment:
This seems arbitrary. Is it that useless to run that test with `image1_headroom == image2_headroom`?

Reply:
Added a comment. It is indeed arbitrary, but the  tone mapping will behave unexpectedly (for the test) if the two values are equal (it will never tone map).

Reply:
> it will never tone map

I meant it could test for that rather than change the input values, but this may already be part of another test.

Reply:
Done in #1876